### DISCUSSION TITLE
Cloning a repo with branch protection rules

### DISCUSSION THREAD

**Discussion by francis-lopes-01** [community/community#36389] [New to GitHub]
I have a bunch of repos to migrate from one organization to another but I am facing the following problem:

when cloning the repo it is not cloning the branch protection rules, so I need to manually set it up again.
Is there any way of replicate the branch configuration while cloning?

**Comment by airtower-luna** [comment]
The simplest solution is probably to *transfer* the repositories instead of recreating them, that should take the settings (and various other things like issues) along. See: https://docs.github.com/en/repositories/creating-and-managing-repositories/transferring-a-repository

Otherwise you should be able to retrieve and set branch protection rules over the API, but it might take a bit of scripting. See: https://docs.github.com/en/rest/branches/branch-protection

**Reply by francis-lopes-01** [reply]
Thank you @airtower-luna! It seams _transfer_ is not an option for us so I may have to go with the API one. I will check it now.

**Comment by ossanna16** [comment]
Hey there @francis-lopes-01 and thank you for asking a good question! I'm glad you found our community 🙂

**Comment by sfonseka-provenir** [comment]
can we have this feature please

**Comment by olyhuta** [comment]
+1 , also need this feature please
